editor/app/src/main/java/ca/ubc/eml/soiltopargraphy/editor/ui/flag/FlagMapFragment.java

```diff
@@ -88,20 +90,26 @@ public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
 
     @Override
     public boolean onOptionsItemSelected (MenuItem item) {
+        FragmentManager manager = getActivity().getSupportFragmentManager();
+        FragmentTransaction transaction = manager.beginTransaction();
```

Comment:
You can reduce some code duplication by having a variable to hold the new fragment then do the transation code after the switch block. In switch, you would just set the variable instead of calling `replace()` and `commit()`.

Something like:
```java
Fragment newFragment = null;
switch (item.getItemId()) {
    case R.id.action_edit:
        newFragment = new DescriptionPanelFragment();
        break;
    case R.id.action_delete:
         mViewModel.deleteFlag();
         newFragment = new FlagListFragment();
         break;
    // ...etc
}
if (newFragment != null) {
    transaction.replace(R.id.container, newFragment);
    transaction.commit();
}
```